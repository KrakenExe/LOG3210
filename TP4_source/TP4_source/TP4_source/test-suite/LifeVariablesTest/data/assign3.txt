Declaration
num a;
num b;
num c;
num d;
EndDeclaration

a = b + 1;
c = a + b;
d = b + d;
a = d + c;