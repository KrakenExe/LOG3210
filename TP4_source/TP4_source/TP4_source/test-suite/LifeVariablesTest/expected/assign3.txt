===== STEP 0 ===== 
REF : [b]
DEF : [a]
IN  : [b, d]
OUT : [a, b, d]
SUCC: [_step1]
PRED: []

===== STEP 1 ===== 
REF : [a, b]
DEF : [c]
IN  : [a, b, d]
OUT : [b, c, d]
SUCC: [_step2]
PRED: [_step0]

===== STEP 2 ===== 
REF : [b, d]
DEF : [d]
IN  : [b, c, d]
OUT : [c, d]
SUCC: [_step3]
PRED: [_step1]

===== STEP 3 ===== 
REF : [c, d]
DEF : [a]
IN  : [c, d]
OUT : []
SUCC: []
PRED: [_step2]

